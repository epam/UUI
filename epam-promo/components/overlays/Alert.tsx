import * as React from 'react';
import * as uui from '@epam/uui';
import { withMods } from '@epam/uui-core';
import { ReactComponent as SuccessIcon } from '../../icons/notification-check_circle-fill-24.svg';
import { ReactComponent as WarningIcon } from '../../icons/notification-warning-fill-24.svg';
import { ReactComponent as ErrorIcon } from '../../icons/notification-error-fill-24.svg';
import { ReactComponent as HintIcon } from '../../icons/notification-help-fill-24.svg';
import { EpamPrimaryColor } from '../types';

export interface AlertMods {
    /** @default 'blue' */
    color?: EpamPrimaryColor;
}

export interface AlertProps extends Omit<uui.AlertProps, 'color'>, AlertMods {}

export const Alert = withMods<Omit<uui.AlertProps, 'color'>, AlertMods>(
    uui.Alert as any, // TODO: need to fix original component
    () => [],
    (props) => ({
        ...props,
        color: props.color ?? 'blue',
    }),
);

export const WarningAlert = React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ WarningIcon } color="amber" ref={ ref } { ...props } />);

export const SuccessAlert = React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ SuccessIcon } color="green" ref={ ref } { ...props } />);

export const HintAlert = React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ HintIcon } color="blue" ref={ ref } { ...props } />);

export const ErrorAlert = React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ ErrorIcon } color="red" ref={ ref } { ...props } />);
