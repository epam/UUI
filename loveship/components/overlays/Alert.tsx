import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';
import { EpamPrimaryColor } from '../types';

interface AlertMods {
    /**
     * Defines component color.
     * @default 'sky'
     */
    color?: EpamPrimaryColor | uui.AlertProps['color'];
}

/** Represents the properties of the Alert component. */
export interface AlertProps extends uui.AlertCoreProps, AlertMods {}

export const Alert = /* @__PURE__ */createSkinComponent<uui.AlertProps, AlertProps>(
    uui.Alert,
    (props) => ({
        ...props,
        color: props.color ?? 'sky',
    }),
);
