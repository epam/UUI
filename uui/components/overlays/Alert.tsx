import * as React from 'react';
import cx from 'classnames';
import { IHasChildren, IHasCX, Icon, IHasRawProps } from '@epam/uui-core';
import { IconContainer } from '@epam/uui-components';
import { IconButton, LinkButton } from '../buttons';
import { ReactComponent as SuccessIcon } from '@epam/assets/icons/notification-check-fill.svg';
import { ReactComponent as WarningIcon } from '@epam/assets/icons/notification-warning-fill.svg';
import { ReactComponent as ErrorIcon } from '@epam/assets/icons/notification-error-fill.svg';
import { ReactComponent as HintIcon } from '@epam/assets/icons/notification-help-fill.svg';
import { ReactComponent as CrossIcon } from '@epam/assets/icons/navigation-close-outline.svg';
import css from './Alert.module.scss';

interface AlertNotificationAction {
    /*
    * Defines notification action name.
    */
    name: string;
    /*
    * Called when action is clicked
    */
    action: () => void;
}

interface AlertMods {
    /** Alert color */
    color: 'info' | 'success' | 'warning' | 'error';
}

export interface AlertCoreProps extends IHasChildren, IHasCX, IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
    /** List of actions to display in the alert. Each action has name and 'action' callback */
    actions?: AlertNotificationAction[];
    /** When specified, a close icon is rendered. onClose callback will be called on clicking the close icon */
    onClose?(): void;
    /** An optional icon to show on the left of the alert */
    icon?: Icon;
    /**
     * Component size. If omitted, 48 size will be used.
     * @default '48'
     */
    size?: '36' | '48';
}

/** Represents the properties of the Alert component. */
export interface AlertProps extends AlertCoreProps, AlertMods {}

export const Alert = /* @__PURE__ */React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
    <div
        role="alert"
        ref={ ref }
        className={
            cx(
                'uui-alert',
                css.root,
                css.alertWrapper,
                props.color && `uui-color-${props.color}`,
                props.cx,
                (props.size === '36' ? css.size36 : css.size48),
            )
        }
        { ...props.rawProps }
    >
        <div className={ css.mainPath }>
            {props.icon && (
                <div className={ css.iconWrapper }>
                    <IconContainer icon={ props.icon } cx={ css.actionIcon } />
                </div>
            )}
            <div className={ css.content }>
                {props.children}
                {props.actions && (
                    <div className={ css.actionWrapper }>
                        {props.actions.map((action) => (
                            <LinkButton caption={ action.name } onClick={ action.action } key={ action.name } cx={ css.actionLink } size={ props?.size === '36' ? '24' : '30' } />
                        ))}
                    </div>
                )}
            </div>
            {props.onClose && <IconButton icon={ CrossIcon } color="neutral" onClick={ props.onClose } cx={ css.closeIcon } />}
        </div>
    </div>
));

export const WarningAlert = /* @__PURE__ */React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ WarningIcon } color="warning" ref={ ref } { ...props } />);

export const SuccessAlert = /* @__PURE__ */React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ SuccessIcon } color="success" ref={ ref } { ...props } />);

export const HintAlert = /* @__PURE__ */React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ HintIcon } color="info" ref={ ref } { ...props } />);

export const ErrorAlert = /* @__PURE__ */React.forwardRef<HTMLDivElement, Omit<AlertProps, 'color'>>((props, ref) => <Alert icon={ ErrorIcon } color="error" ref={ ref } { ...props } />);
