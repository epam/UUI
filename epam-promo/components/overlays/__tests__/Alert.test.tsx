import React from 'react';
import { Alert, ErrorAlert, HintAlert, SuccessAlert, WarningAlert } from '../Alert';
import renderer from 'react-test-renderer';
import { ReactComponent as CalendarIcon } from '../../../icons/calendar-18.svg';

describe('Alert', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<Alert />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', () => {
        const tree = renderer
            .create(
                <Alert
                    icon={CalendarIcon}
                    color="amber"
                    actions={[
                        {
                            name: 'ACTION 1',
                            action: jest.fn,
                        },
                    ]}
                    onClose={jest.fn}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('WarningAlert', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<WarningAlert />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('SuccessAlert', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<SuccessAlert />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('HintAlert', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<HintAlert />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('ErrorAlert', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<ErrorAlert />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
