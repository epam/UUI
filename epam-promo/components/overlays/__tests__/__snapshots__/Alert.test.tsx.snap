// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should be rendered correctly 1`] = `
<div
  className="alertWrapper color-blue root"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="content"
    />
  </div>
</div>
`;

exports[`Alert should be rendered correctly 2`] = `
<div
  className="alertWrapper color-amber root"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    >
      <div
        className="actionWrapper"
      >
        <button
          className="container uui-button-box uui-enabled -clickable root size-24 linkButton-color-blue uui-no-left-icon uui-no-right-icon actionLink"
          href={null}
          onClick={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <div
            className="uui-caption"
          >
            ACTION 1
          </div>
        </button>
      </div>
    </div>
    <button
      className="container uui-button-box uui-enabled -clickable root icon-color-gray60 closeIcon"
      href={null}
      onClick={[Function]}
      role="button"
      tabIndex={0}
      type="button"
    >
      <div
        className="container uui-icon uui-enabled"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`ErrorAlert should be rendered correctly 1`] = `
<div
  className="alertWrapper color-red root"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
  </div>
</div>
`;

exports[`HintAlert should be rendered correctly 1`] = `
<div
  className="alertWrapper color-blue root"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
  </div>
</div>
`;

exports[`SuccessAlert should be rendered correctly 1`] = `
<div
  className="alertWrapper color-green root"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
  </div>
</div>
`;

exports[`WarningAlert should be rendered correctly 1`] = `
<div
  className="alertWrapper color-amber root"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
  </div>
</div>
`;
